src/test/addrman_tests.cpp

```diff
@@ -444,10 +444,20 @@ BOOST_AUTO_TEST_CASE(getaddr_unfiltered)
     CNetAddr source = ResolveIP("250.1.2.1");
     BOOST_CHECK(addrman->Add({addr1, addr2}, source));
 
+    // Set time on this addr so isTerrible = false
+    CAddress addr3 = CAddress(ResolveService("250.251.2.3", 9998), NODE_NONE);
+    addr3.nTime = Now<NodeSeconds>();
+    addrman->Good(addr3, /*time=*/Now<NodeSeconds>());
+    BOOST_CHECK(addrman->Add({addr3}, source));
+    // The time is set, but after 3 unsuccessful attempts this addr should be isTerrible = true
+    for (size_t i = 0; i < 3; ++i) {
+        addrman->Attempt(addr3, /*fCountFailure=*/true, /*time=*/Now<NodeSeconds>() - 10h);
```

Comment:
The time setting here seems to be critical and could be commented.  It looks like the threshold between success and failure is between `60s` and `61s`. Is that the case for you as well, and could you explain here why?

Reply:
Note that `IsTerrible` has other verifications (e.g. we don't remove things tried in the last minute). So the timing setting here is to explicity reach exactly what we want to test but could be any other value. Every time we call `Attempt`, `m_last_try` is updated. So, yes, you're right about the threshold. Worth adding a comment?

Reply:
Oh indeed (thanks!) I think the following would be clearer.

```diff
@@ -453,12 +453,13 @@ BOOST_AUTO_TEST_CASE(getaddr_unfiltered)
     addr3.nTime = Now<NodeSeconds>();
     addrman->Good(addr3, /*time=*/Now<NodeSeconds>());
     BOOST_CHECK(addrman->Add({addr3}, source));
-    // The time is set, but after 3 unsuccessful attempts this addr should be isTerrible = true
+    // The time is set, but after ADDRMAN_RETRIES unsuccessful attempts not
+    // retried in the last minute, this addr should be isTerrible = true
     for (size_t i = 0; i < 3; ++i) {
-        addrman->Attempt(addr3, /*fCountFailure=*/true, /*time=*/Now<NodeSeconds>() - 10h);
+        addrman->Attempt(addr3, /*fCountFailure=*/true, /*time=*/Now<NodeSeconds>() - 61s);
     }
 
-    // Filtered GetAddr should only return addr1
+    // GetAddr filtered by quality (i.e. not IsTerrible) should only return addr1
     BOOST_CHECK_EQUAL(addrman->GetAddr(/*max_addresses=*/0, /*max_pct=*/0, /*network=*/std::nullopt).size(), 1U);
     // Unfiltered GetAddr should return all addrs
     BOOST_CHECK_EQUAL(addrman->GetAddr(/*max_addresses=*/0, /*max_pct=*/0, /*network=*/std::nullopt, /*filtered=*/false).size(), 3U);
```

Reply:
Done, thanks.